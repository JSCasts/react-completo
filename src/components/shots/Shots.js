import React, { Component } from 'react';

import Shot from '../shot/Shot';
import getShots from '../../services/shots';

class Shots extends Component {
  constructor(props) {
    super(props);

    this.state = {
      shots: [],
    };
  }

  componentWillMount() {
    getShots().then(response => this.setState({ shots: response.data }));
  }

  render() {
    return (
      <div>
        { this.state.shots.map(shot =>
          <Shot
            key={shot.id}
            id={shot.id}
            image={shot.images.normal}
            date={shot.created_at}
            title={shot.title}
            description={shot.description}
            viewsCount={shot.views_count}
            likesCount={shot.likes_count}
          />)
        }
      </div>
    );
  }
}

export default Shots;
