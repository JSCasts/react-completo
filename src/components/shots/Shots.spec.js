import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Shots from './Shots';

test('Snapshot', () => {
  let component = renderer.create(
    <Shots />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Shots', () => {
  test('has state', () => {
    const component = shallow(
      <Shots />
    );
    expect(component.state()).toEqual({ 
      page: 1,
      perPage: 16,
      shots: [], 
    });
  });

  describe('when state is null', () => {
    test('does not contain Shots', () => {
      const component = shallow(
        <Shots />
      );
      expect(component.text()).not.toContain('Shot');
    });
  });

  describe('when state is defined', () => {
    test('does not contain Shots', () => {
      const component = shallow(
        <Shots />
      );
      const shot = {
        id: 1,
        created_at: '2010-01-01',
        images: {
          teaser: 'teaser',
        },
        title: 'title',
        description: 'description',
        views_count: 13,
        likes_count: 14,
      };
      component.setState({ shots: [ shot ]});

      expect(component.text()).toContain('Shot');
      expect(component.html()).toContain(shot.id);
      expect(component.html()).toContain(shot.created_at);
      expect(component.html()).toContain(shot.images.teaser);
      expect(component.html()).toContain(shot.views_count);
      expect(component.html()).toContain(shot.likes_count);
    });
  });

  describe('onScroll', () => {
    test('updates the page', () => {
      const component = shallow(
        <Shots />
      );
      window.innerHeight = 0;
      window.scrollY = 0;
      document.body.scrollHeight = 1;
      component.instance().onScroll();
      expect(component.state().page).toEqual(1);

      window.innerHeight = 1;
      window.scrollY = 1;
      document.body.scrollHeight = 0;
      component.instance().onScroll();
      expect(component.state().page).toEqual(2);
    });
  });
});

