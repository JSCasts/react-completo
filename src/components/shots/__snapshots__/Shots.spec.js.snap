exports[`test Snapshot 1`] = `<div />`;
